### DISCUSSION TITLE
Establishing a community fork of orgzly

### DISCUSSION THREAD

**Discussion by markamaze** [orgzly-revived/orgzly-android-revived#3] [General]
It seems to me that before any development is done, the new repo needs to be established. There isn't much point in updating the code if there isn't an easy way for users to use it, and there is almost no knowledge of it existing. 

Off the top of my head, these things should be done if this fork of orgzly is too be successful:

- There needs to be a new app available for download on play store and fdroid

- the Readme needs to be updated to reflect it is forked from original and update links to download new version (plus credit given to the creator, who hasn't been heard from which is why it's being forked)

- the orgzly community needs to know about the transition

- disconnect from orgzly.com due to lack of access, implement docs in another way

**Comment by alensiljak** [comment]
While these are elements needed for a successful project, I don't necessarily agree that they need to be done before any development. Quite the contrary.
For me it is more important to get certain features and functionality in the app than what the app is called or where the documentation is. I think I only looked at the Orgzly documentation once, to see the search syntax.

If there is enough manpower then sure, why not. But if not, then we might as well add the missing features and publish the build on GitHub until all those side activities are complete. I've witnessed how long it takes to publish an app on F-Droid and Play Store. I just fear that, if we wait until all those issues are resolved, this fork may not offer more than Orgzly for a fairly long time.

**Reply by markamaze** [reply]
I see your point and don't think no other work could be done as not everyone can work only on my suggestions. I'm sure some will find this fork and manually build and install it, but I guess very few orgzly users will. In that case any updated code will have very little reach.

I suppose all this depends on the goal of this fork of orgzly. My assumption was, the goal is to reproduce orgzly in order for development to continue since those interested in doing so don't have access. Reproducing it means to me, making it available for anyone to easily download and use on an android device, while volunteer developers can contribute to the maintenance and enhancement of it. 

My thinking is the priority should be to reestablish what exists and get people to switch while they haven't moved on yet. That requires providing a new app - which requires a name, so that's a good opportunity to discuss what that should be exactly - and updating the info about the project. If the usability of this fork by the average orgzly user is put off, they will eventually move on and it will be much harder to rebuild the community of users and contributors. In that case this will just become a project for a handful of developers who want it for themselves. 

Maybe my assumptions are wrong though. It wouldn't be the first time.

**Comment by amberin** [comment]
I agree that getting development going is important, but so far not much has happened, and I agree with @markamaze on this:

> My thinking is the priority should be to reestablish what exists and get people to switch while they haven't moved on yet. That requires providing a new app - which requires a name, so that's a good opportunity to discuss what that should be exactly - and updating the info about the project. If the usability of this fork by the average orgzly user is put off, they will eventually move on and it will be much harder to rebuild the community of users and contributors.

**Comment by amberin** [comment]
I'm thinking priority 1 should be making an APK artifact available via Github, priority 2 to look at F-Droid publishing, at least getting the ball rolling in some sense.

**Comment by karlicoss** [comment]
Agree it's worth prioritizing publishing on play/f-droid, the app thankfully works as it is (of course there are some bugs, but I believe nothing major). But with another android release it's possible that the app could be at risk of being completely broken due to permissions shenanigans or something like that (e.g. like happened with storage in Android 12). And if users won't be able to use it, they will just uninstall and forget completely.

Another thing worth putting in readme is how to migrate off the old version of Orgzly 
- for contents: I've always used Orgzly in the "external repository" mode -- but not sure if the user doesn't, are they stored on SD card or in the private app data? Since in the latter case they will just disappear when app is removed?
- for settings: I imagine it'll basically be a matter of setting up from scratch? Perhaps worth stressing that users remember their saved searches, and certain settings can be annoying if they are forgotten (e.g. the ones impacting formatting).

**Comment by erikmd** [comment]
Let me also suggest that Orgzly-android-revived should probably be mentioned "upstream" in:
* [orgmode.org: 20. Org-mode on mobile devices](https://orgmode.org/worg/org-faq.html#org76ef962)

**Reply by yantar92** [reply]
I have been watching this fork evolving. No slowdown, active. Updated Org FAQ now.

**Reply by amberin** [reply]
Please note that www.orgzlyrevived.com now also exists as a "fork" of www.orgzly.com.
